### PR TITLE
oop-programs_with_interesting_classes-easy-q7

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Please see my feedback on other questions in this category and apply it to this PR.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
See Sam's review

File: src/oop/programs_with_interesting_classes/easy/q7/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                {"Shape of You", 3.88},
+                {"Smells Like Teen Spirit", 5.02},
+                {"Rolling in the Deep", 3.80},
+                {"Imagine", 3.07},
+                {"Superstition", 4.43}
+        };
+        for(int i = 0; i < tests.length; i ++) {
+            Object classInstance = testClass2.createInstance();
+            Song[] songs = new Song[i + 1];
+            for(int j = 0; j <= i; j ++) songs[j] = new Song((String)tests[i][0], (double)tests[i][1]);
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q7/MainTest.java:209]: please put this on the next line for easier readability

File: src/oop/programs_with_interesting_classes/easy/q7/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            songs[t ++] = new Song((String)tests[i][0], (double)tests[i][1]);
+            Object[][] arguments = new Object[][]{
+                    {new Song((String)tests[i][0], (double)tests[i][1]), Song.class}
+            };
+            testClass2.callMethod(methodName2, arguments, classInstance);
+            String incorrectNumDays = "The %s method in the %s class does not correctly add to the %s attribute.";
+            Song[] ans = (Song[])testClass2.getFieldValue(classInstance, firstAttributeName2);
+            for(int j = 0; j < songs.length; j++){
+                if(songs[j] != null && ans[j] != null)
+                    _assertEquals(songs[j].toString(), ans[j].toString(), String.format(incorrectNumDays, methodName2, className2, firstAttributeName2));
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q7/MainTest.java:261]: this could fail if the toString is not in place even if the method works

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Pull request merged**